share/disambiguation.py

```diff
@@ -131,9 +134,12 @@ def _instance_for_node(self, node):
                 return None
             if len(found) == 1:
                 return found[0]
-            if all_query.children or any('__' in str(query) for query in queries):
+            if all_query.children:
                 logger.warning('Multiple %ss returned for %s (The main query) bailing', concrete_model, all_query)
                 break
+            if all('__' in str(query) for query in queries):
```

Comment:
Why is this changed to `all`?  If we have 2 identifiers and a name this case won't be hit.